scripts/fdbshow

```diff
@@ -0,0 +1,192 @@
+            except KeyError as e:
+                continue
+            
+            self.bridge_mac_list.append((int(fdb["vlan"]),) + (fdb["mac"],) + (if_name,))
+
+        self.bridge_mac_list.sort(key = lambda x: x[0])
+        return
+    
+    
+    def get_iter_index(self, key_value=0, pos=0):
+        """
+            Get the starting index of matched entry
+        """
+        if pos == 0:
+            self.bridge_mac_list.sort(key = lambda x: x[pos])
+        keys = [r[pos] for r in self.bridge_mac_list]
+        try:
+            return keys.index(key_value)
+        except IndexError as e:
+            print e.message
+
+
+    def display_specific(self, port, vlan):
+        """
+            Display the FDB entries for specified vlan/port.
+        """
+        output = []
+
+        if vlan is not None and port is None:
+            vlan = int(vlan)
+            s_index = self.get_iter_index(vlan)
+            fdb_list = [fdb for fdb in self.bridge_mac_list[s_index:] if fdb[0] == vlan]
+
+        elif vlan is None and port is not None:
+            self.bridge_mac_list = natsorted(self.bridge_mac_list, key = lambda x: x[2])
+            s_index = self.get_iter_index(port, 2)
+            fdb_list = [fdb for fdb in self.bridge_mac_list[s_index:] if fdb[2] == port]
+
+        else:
+            vlan = int(vlan)
+            s_index = self.get_iter_index(vlan)
+            fdb_list = [fdb for fdb in self.bridge_mac_list[s_index:]
+                        if fdb[0] == vlan and fdb[2] == port]
+
+        for fdb in fdb_list:
+            self.FDB_COUNT += 1
+            output.append([self.FDB_COUNT, fdb[0], fdb[1], fdb[2]])
+
+        print tabulate(output, self.HEADER)
+        print "Total number of entries {0} ".format(self.FDB_COUNT)
+
+
+    def display_summary(self):
+        """
+            Display the FDB summary. 
+            @todo: - PortChannel support
+        """
+        output = []
+        self.bridge_mac_list.sort(key = lambda x: x[0])
+        for fdb in self.bridge_mac_list:
+            self.FDB_COUNT += 1
+            output.append([self.FDB_COUNT, fdb[0], fdb[1], fdb[2]])
+            
+        print tabulate(output, self.HEADER)
+        print "Total number of entries {0} ".format(self.FDB_COUNT)
+        
+        
+def main():
+    
+    parser = argparse.ArgumentParser(description='Display ASIC FDB entries',
+                                     formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument('-p', '--port', type=str, help='FDB learned on specific port: Ethernet0', default=None)
+    parser.add_argument('-v', '--vlan', type=str, help='FDB learned on specific Vlan: 1001', default=None)
+    args = parser.parse_args()
+
+    try:
+        fdb = FdbShow()
+        if args.port is None and args.vlan is None:
+            fdb.display_summary()
+        else:
+            fdb.display_specific(args.port, args.vlan)
+        
+    except Exception as e:
```

Comment:
If you are arguing about this case, the solution should be catch specific exception, instead of catch all.

---
In reply to: [140927960](https://github.com/Azure/sonic-utilities/pull/106#discussion_r140927960) [](ancestors = 140927960)